### ISSUE TITLE
Tooltip Issue - WoW Classic

### ISSUE THREAD

**Issue by Vequis-uk** [brittyazel/Neuron#354]
Hey,

This addon is perfect but I have some issue in classic with tooltips.

For example on Mage Portals the quantity isnt shown for the amount of Rune of Portals in inventory.

The same for warlocks with pets, the amount of soulshards are not shown, regardless of #showtooltip being present or not.

If it would be possible to look into this, it would be greatly appreciated!

Regards,
David

**Comment by crosenbe** [comment]
Consumables are shown only a second and disappear.

**Comment by Vequis-uk** [comment]
It did work when classic first launched so I suspect Blizzard changed something

**Comment by crosenbe** [comment]
Maybe, enchanced tooltips are not working only, just mentioned.

**Comment by crosenbe** [comment]
Is this project discontinued?

**Comment by AlexFolland** [comment]
No.  What makes you think that?  It was last updated less than 2 months ago and works well.  Also, it's FOSS, so if you have something to contribute, you are free to fork and submit a pull request.  There's plenty of life here.

**Comment by crosenbe** [comment]
Continious commits until beginning of May, from May no activity. Please don't understand me wrong, it seems the work on this Addon stopped because the activity changed. It's not a request do work on topics, just to be sure if it was discontinued or not.

**Comment by brittyazel** [comment]
> Is this project discontinued?

Uh no? I mean I have an active branch that's a near full rewrite of the project that I pushed code to just a couple weeks ago. I haven't been putting out new releases as the latest version is decently stable as-is and I've been focusing all my time on the new development branch for what will be Neuron 2.0. And, more specifically the only reason I haven't been pushing code these last couple weeks are A) my other add-on Enhanced Raid Frames needed some attention and love, and B) I'm really swamped at work and getting my PhD and I'm trying to stay afloat  in those respects.

So no, this project is not abandoned, and it fully up to date with modern content, and the dev branch is thriving with changes building up to what will be a monumental 2.0 release sometime in the next few months. Probably around the time that Shadowlands hits. I'm always looking for help though. Maintaining multiple add-ons and collectively tens of thousands of lines of code all by myself in my off-time is a ton of work, so things sometimes move slowly.

**Comment by crosenbe** [comment]
I would try to work a bit more but i didn't find any documentation about the API of Blizz. No official and no community driven which seems to be complete.

**Comment by brittyazel** [comment]
Wowpedia is your best bet:
https://wow.gamepedia.com/Global_functions

I also linked a pretty comprehensive guide to setting up a good add-on development environment in the readme. I use IntelliJ IDEA + Emmylua and it is fantastic. I also did a bunch of work rewriting the Ellypse API documentation generation scripts over on his GitHub repo, so that's taken some time.

In general, going 2 months without changes in FOSS, volunteer projects is pretty standard. I'd love to write patches for all my add-ons every day, but it's hard to juggle this + work + school + my fiance + my health + actually playing the damn game, all alone. 

But yes, the NeuronNext branch is thriving, stable, and I'm just working on the GUI (which I absolutely loathe). I love backend work, but I'm suffering through the frontend side out of the goodness of my heart. Feel free to try out that branch and tell me what you think, though note it has a slightly different database structure, there's a bunch of new bar types, and the GUI is only 40% implemented.

**Comment by AlexFolland** [comment]
Great response @brittyazel though I think you shouldn't have to defend yourself.  Can there be a rule to not pester volunteer authors of free open-source software about progress?  I'm glad that is so widely-understood on github.

Of course there's API documentation.  https://wow.gamepedia.com/World_of_Warcraft_API is extremely comprehensive, and there's a mirror of the WoW UI source code so you can see how Blizzard implements the default UI here: https://github.com/Gethe/wow-ui-source .  Also there are incredibly comprehensive information sources like https://www.townlong-yak.com/framexml/live/Blizzard_APIDocumentation , the other tools at Townlong Yak, and https://wow.tools/ .  If you need more help, you can visit the #wowuidev IRC channel on freenode.

**Comment by brittyazel** [comment]
@AlexFolland There's also a really good add-on author discord channel!

Just so you all know, I was also serving on the Board of Directors of a free software non-profit foundation until just a few weeks ago, but my term just ended so I now have more time for hobbies. My two graduate courses that I teach are also done for the summer, so I have a good chunk more free time. I've just recently taken up Enhancement Shaman for M+ though, so I do have a new side hustle 😉

**Comment by crosenbe** [comment]
@brittyazel Thanks a lot providing the links.
@AlexFolland I think you misinterpreted my initial question, i just asked if this project is still alive, i didn't push for activity or i didn't ask to work on something. A simple yes or no is fair enough. Sometimes i don't understand why people think the reason of the question is to tackle people, it's not like that. I hope this is understood in the right way now.

**Comment by AlexFolland** [comment]
What you said is still pestering about progress, whether you intended it or not.  Think about it from their perspective.  "Is it discontinued?" pretty much forces an author to defend their project, to avoid rumors spreading that their project is dead and no longer worth considering, which is pretty disheartening.  There are better ways to discover it, like clicking the author's name, finding when they last committed, and seeing what the commits were.  It's generally better to never ask if a FOSS project is active, because unless it's removed, it's active, even if the activity happens 10 years later than you'd expect.  FOSS means anyone can pick it up, and this particular project is a great example of that.  It has had 3 excellent primary maintainers over the years, who have each put tons of effort into it: Maul, SLOknigtfall, and Britt here.  All of them deserve respect (no matter what else they've done) for putting their free time in to keep this awesome project running as Blizzard mercilessly changes the WoW UI API.

**Comment by brittyazel** [comment]
It's fine man, I'm not offended and @AlexFolland is just looking out for me.

In general, there's a trend in FOSS projects for users to get aggressive and demanding about the pace of work from the developers. Though, to be clear, I'm not saying that's what you did. I have had users scream at me over bugs, call me trash, demand I fix something or add a feature that they need, complain that I'm too slow, etc, which is largely considered bad form to do to a volunteer developer. Lots of devs have gone the defensive route to protect themselves, so that's what you may be seeing from @AlexFolland

**Comment by Vequis-uk** [comment]
Heyi have only just seen these comments, I have used Neuron for a very long time and am really looking forward to the 2.0 update. Works 99% perfect for me on Classic and still imo the best action bar addon. Keep up the good work 😊👍

**Comment by Vequis-uk** [comment]
And yes, these projects are massive work, I think twitch should provide some sort of funding even if it's small. Would certainly be a nice gesture!

**Comment by AlexFolland** [comment]
There is the CurseForge author rewards thing, but it's laughably small (like fractions of a cent per month), and I found out recently that I'm not even eligible to cash out even though I now have $35 because they changed their requirements since I started putting addons on CurseForge 7 years ago.  I'd rather spend that $35 on not having to meet those requirements.

However, there is GitHub Sponsors, which is enabled for this repository.  One could sponsor @brittyazel by clicking this link: https://github.com/sponsors/brittyazel

**Comment by crosenbe** [comment]
I tried to figure it out but without any success. It's clear only item tooltips are affected and only with extended tooltips. Because extended tooltips are working with Bartender i compared the included libraries, but i didn't find any difference. Also the update function of the tooltip works as expected. 
I disabled all other addons to identify if the issue came from addons changing the extended tooltip. Still the same issue. the only difference is the configured anchor, but even with using

`GameTooltip_SetDefaultAnchor(GameTooltip, self);
`
instead of 

`GameTooltip:SetOwner(self, "ANCHOR_RIGHT")`

it does not work. Any additional ideas?
